packages/parser/src/actions/CombatSupportAction.ts

```diff
@@ -44,56 +32,52 @@ export class CombatSupportAction extends CombatAction {
   constructor(logLine: ILogLine, wowVersion: WowVersion) {
     super(logLine);
     if (!CombatSupportAction.supports(logLine)) {
-      throw new Error('Event not supported as CombatAdvancedAction: ' + logLine.raw);
+      throw new Error('Event not supported as CombatSupportAction: ' + logLine.raw);
     }
 
-    const advancedLoggingOffset = logLine.event.startsWith('SWING_') ? 8 : 11;
+    /**
+     * For some god forsaken reason blizzard has decided that these _SUPPORT events will actually drop fields instead
+     * of zeroing them in non-advanced logging modes. This means our prior strategy of just ingesting the zero/nil
+     * fields won't work because it will crash with out of index errors. This is the only one I've seen in the new style
+     * so I am just casing it out here.
+     */
+    if (logLine.event === 'SWING_DAMAGE_LANDED_SUPPORT' && logLine.parameters.length == 22) {
+      this.amount = logLine.parameters[11];
 
-    this.advanced = logLine.parameters[advancedLoggingOffset] !== 0;
-    this.advancedActorId = logLine.parameters[advancedLoggingOffset].toString();
-    this.advancedOwnerId = logLine.parameters[advancedLoggingOffset + 1].toString();
-    this.advancedActorCurrentHp = logLine.parameters[advancedLoggingOffset + 2];
-    this.advancedActorMaxHp = logLine.parameters[advancedLoggingOffset + 3];
+      this.isCritical = false; // it's nil in the log
+      if (getUnitType(this.destUnitFlags) === CombatUnitType.Player) {
+        this.effectiveAmount = -this.amount;
+      } else {
+        this.effectiveAmount = 0;
+      }
+      this.supportActorId = logLine.parameters[21];
+      return;
+    }
 
     const wowVersionOffset = wowVersion === 'retail' ? 0 : -1;
 
-    const powerType = logLine.parameters[advancedLoggingOffset + wowVersionOffset + 8]
```

Comment:
all these logic no longer needed?

Reply:
yeah none of this stuff is needed and just ends up adding a bunch of fields that are hard to understand (since there are 3 actors in each of these events); the _SUPPORT events are all just adding information about how to attribute damage, not actually describing hp/mp changes since we are parsing the actual events that are doing the 'work' so to speak